### PR TITLE
Bump FAST API upper bounds

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

- This PR bumps FastAPI upper bound.
- This lays foundation to support a migration from pydantic v1 to v2 (Kedro Viz still uses pydantic v1)
- We are pinning the pydantic version as we are not able to use pydantic.v1 api when pydantic v2 is installed. 

Open issue - https://github.com/tiangolo/fastapi/issues/10360

## Development notes

- Updated requirement files to reflect Fast API upgrade

## QA notes

- All test cases should pass
- kedro viz should visualize flowchart without any issue

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: package/requirements.txt [hunk 1]
```diff
@@ -1,7 +1,8 @@
 packaging~=23.0
 kedro>=0.17.5
 ipython>=7.0.0, <9.0
-fastapi>=0.73.0, <0.96.0
+fastapi>=0.100.0,<0.104.0
```
Comment (rashidakanchwala) [inline package/requirements.txt:4]:
```suggestion
fastapi>=0.73.0
```

can we just remove the upper bound

**Review by rashidakanchwala — APPROVED** [review]

File: package/requirements.txt [hunk 2]
```diff
@@ -1,7 +1,9 @@
 packaging~=23.0
 kedro>=0.17.5
 ipython>=7.0.0, <9.0
-fastapi>=0.73.0, <0.96.0
+fastapi>=0.73.0,<0.200.0; python_version >= '3.8'
+fastapi>=0.73.0,<0.104.0; python_version < '3.8'
```
Comment (astrojuanlu) [inline package/requirements.txt:5]: Is Python < 3.8 still supported?

**Review by astrojuanlu — COMMENTED** [review]
Looks good to me!

File: package/requirements.txt (hunk 2 above)
Comment (rashidakanchwala) [reply package/requirements.txt:5]: yes, plan was to drop 3.7 once framework had but looks like you already have? We will plan in the upcoming sprint.

**Pull request merged**